docs: add an example for Yarn 2+ (#11696)
renovatebot/renovate: docs/usage/getting-started/private-packages.md

Renovate updates `npmRegistries` in `.yarnrc.yml` with resolved `hostRules` before running Yarn.

> For Renovate to overwrite existing `npmRegistries` entry, the key should match the `matchHost` without the protocol (`http:` or `https:`) and with the trailing slash.

Author: ```suggestion
For Renovate to overwrite existing `npmRegistries` entry, the key should match the `matchHost` minus the protocol (`http:` or `https:`) plus the trailing slash.
```
Would this be more clear?

Reviewer: An example would be ideal

Author: The example below is for this.

Reviewer 2: Can you also add the hostrule to the sample.

I thought this sentence is about the hostrule and not the `.yarnrc.yaml` file. 🤔

Author: It refers to the example directly above this, but to be more clear 😄

For example, Renovate configuration above this section will update `.yarnrc.yml` as following:

```yaml
npmRegistries:
  //npm.pkg.github.com/:
    npmAuthToken: <Decrypted PAT Token>
  //npm.pkg.github.com:
    # this will not be overwritten
  https://npm.pkg.github.com/:

>     # this will not be overwritten

Author: My intention was as they are valid Yarn configuration, they would conflict with the configuration set by Renovate, so the last sentence was to clear them.

Renovate updates `npmRegistries` in `.yarnrc.yml` with resolved `hostRules` before running Yarn.
For Renovate to overwrite existing `npmRegistries` entry, the key should match the `matchHost` without the protocol (`http:` or `https:`) and with the trailing slash.

For example, the Renovate configuration:

```json
{
  "hostRules": [
    {
      "matchHost": "https://npm.pkg.github.com/",
      "hostType": "npm",
      "encrypted": {
        "token": "<Encrypted PAT Token>"
      }
    }
  ]
}
```

will update `.yarnrc.yml` as following:

```yaml
npmRegistries:
  //npm.pkg.github.com/:
    npmAuthToken: <Decrypted PAT Token>
  //npm.pkg.github.com:
    # this will not be overwritten and may conflict
  https://npm.pkg.github.com/:

>     # this will not be overwritten and may conflict

Author: My intention was that they would not be overwritten, but as they are valid Yarn configuration, it'd conflict with the configuration set by Renovate, so the the last sentence was to prevent it.

Reviewer: Should we change our behavior or do you think it's ok?

Author: I don't have a strong opinion. The behavior can be changed here, though: https://github.com/renovatebot/renovate/blob/62a540c272d4f0b2738781d0442d61fd946cb482/lib/manager/npm/post-update/index.ts#L519-L522

Reviewer: Let's merge this docs update though as it's useful already
